src/viewProviders/flinkArtifacts.test.ts

```diff
@@ -83,79 +85,144 @@ describe("FlinkArtifactsViewProvider", () => {
       assert.deepStrictEqual(viewProvider["_artifacts"], mockArtifacts);
     });
 
-    it("should handle 401 auth errors and fire auth session invalidated", async () => {
-      const windowWithProgressStub = sandbox
-        .stub(window, "withProgress")
-        .callsFake((_, callback) => {
-          // Call the callback immediately with a resolved promise
-          const mockProgress = {} as Progress<unknown>;
-          const mockToken = {} as CancellationToken;
-          return Promise.resolve(callback(mockProgress, mockToken));
-        });
-
-      const authInvalidatedFireStub = sandbox.stub(ccloudAuthSessionInvalidated, "fire");
-      const resource = TEST_CCLOUD_FLINK_COMPUTE_POOL;
-      viewProvider["resource"] = resource;
+    it("returns artifacts when compute pool is selected", () => {
+      const mockArtifacts = [
+        createFlinkArtifact({
+          id: "artifact1",
+          name: "Test Artifact 1",
+        }),
+      ];
 
-      const stubbedLoader: sinon.SinonStubbedInstance<CCloudResourceLoader> =
-        getStubbedCCloudResourceLoader(sandbox);
+      viewProvider["resource"] = TEST_CCLOUD_FLINK_COMPUTE_POOL;
+      viewProvider["_artifacts"] = mockArtifacts;
+
+      const children = viewProvider.getChildren();
+      assert.deepStrictEqual(children, mockArtifacts);
+    });
 
-      const authError = createResponseError(401, "Unauthorized", "test");
-      stubbedLoader.getFlinkArtifacts.rejects(authError);
+    describe("error handling", () => {
+      let stubbedLoader: sinon.SinonStubbedInstance<CCloudResourceLoader>;
 
-      await assert.rejects(async () => {
-        await viewProvider.refresh();
+      beforeEach(() => {
+        viewProvider["resource"] = TEST_CCLOUD_FLINK_COMPUTE_POOL;
+        stubbedLoader = getStubbedCCloudResourceLoader(sandbox);
       });
 
-      sinon.assert.calledOnce(windowWithProgressStub);
-      sinon.assert.calledOnce(stubbedLoader.getFlinkArtifacts);
-      sinon.assert.calledOnce(authInvalidatedFireStub);
-      assert.deepStrictEqual(viewProvider["_artifacts"], []);
-    });
+      it("should handle 4xx HTTP errors with appropriate message", async () => {
+        const mockError = {
+          response: {
+            status: 403,
+            statusText: "Forbidden",
+          },
+        };
+        stubbedLoader.getFlinkArtifacts.rejects(mockError);
+
+        sandbox.stub(errors, "isResponseError").returns(true);
```

Comment:
I'd recommend using ResponseError directly for these tests so you don't have to stub the behavior of `isResponseError()`
```suggestion
        const mockError = new ResponseError(
          new Response("Forbidden", { status: 403 }
        );
        stubbedLoader.getFlinkArtifacts.rejects(mockError);
```